### PR TITLE
Refactor gilded rose and add support for conjured item

### PR DESCRIPTION
**Pull request by rizaljamhari**
- Refactored current implementation by using OOP to eliminate `if...elseif....else` complexity
- Add support for the new conjured item
- Add and update unit test

### CODE REVIEW

File: lib/aged_brie_handler.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    # early return if quality is already at max
+    return if is_max_quality(item.quality)
+
+    if (item.sell_in > 0)
+      quality = item.quality + 1
+    else
+      quality = item.quality + 2
+    end
+
+    item.quality = quality if is_at_most_quality(quality)
```
Comment (renlstr-dev) [inline lib/aged_brie_handler.rb:19]: discuss

File: lib/backstage_passes_handler.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      quality = item.quality + 1
+    elsif item.sell_in > 5
+      quality = item.quality + 2
+    elsif item.sell_in > 0
+      quality = item.quality + 3
+    else
+      quality = 0
+    end
+
+    quality = QualityValidation::MAX_QUALITY if !is_at_most_quality(quality)
```
Comment (renlstr-dev) [inline lib/backstage_passes_handler.rb:20]: thinking if there is a better way to name this

File: lib/modules/quality_validation.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
+module QualityValidation
+  MAX_QUALITY = 50
+  MIN_QUALITY = 0
```
Comment (renlstr-dev) [inline lib/modules/quality_validation.rb:3]: love that you put constant for magic numbers

File: spec/quality_validation_spec.rb [hunk 4]
```diff
@@ -0,0 +1,68 @@
+require "../lib/modules/quality_validation.rb"
+
+describe QualityValidation do
+  include QualityValidation
```
Comment (renlstr-dev) [inline spec/quality_validation_spec.rb:4]: thanks for adding a unit test for the validation

**Review by renlstr-dev — COMMENTED** [review]
i like the fact that you used
- design pattern
- the code is easy to reason with now
- very easy to get context
- +1 on adding a unit test for the validation

File: lib/aged_brie_handler.rb (hunk 1 above)
Comment (rizaljamhari) [reply lib/aged_brie_handler.rb:19]: `item.quality = quality if is_at_most(quality)`

Comment (renlstr-dev) [reply lib/aged_brie_handler.rb:19]: `item.quality = quality if less_than_max_quality?(quality)`

**Pull request closed without merging**